### ISSUE TITLE
Raven always reports errors as being in "handleError" or "handleFatalError"

### ISSUE THREAD

**Issue by mnpenner** [getsentry/sentry-php#343]
See screenshots below. The error clearly happened in `main()` but the description always says `handleError` because that's the name of the function in `Raven_ErrorHandler`.

![image](https://cloud.githubusercontent.com/assets/1112998/18218021/e7f52486-7114-11e6-8c4d-9e494f70a34f.png)

![image](https://cloud.githubusercontent.com/assets/1112998/18218072/3d995ed4-7115-11e6-9841-d17c31699711.png)

Would be nice if this reported correctly.

**Comment by dcramer** [comment]
We could def improve this but you want to set your app root (see config docs) as that will improve stacks.

**Comment by mnpenner** [comment]
You mean `app_path`? I did already.

**Comment by dcramer** [comment]
@mnpenner I would have expected ServerConfig.php (assuming thats your code) to be annotated in this case. Is this stacktrace entirely third party code?

I do think that outside of this there might still be a concern with how we're generating the culprit in PHP.

**Comment by mnpenner** [comment]
What do you mean by "annotated"? 

`ServerConfig` is my code. I set `app_path` to `/usr/local/webenginex`.

**Comment by dcramer** [comment]
We split frames between app and "system" (non-app) when the data is available:

![screenshot 2016-09-06 11 48 28](https://cloud.githubusercontent.com/assets/23610/18286475/da98f504-7427-11e6-9a03-38aa2a63f824.png)

**Comment by mnpenner** [comment]
That's a nice feature. I wonder why it's not working?

Is the splitting done in the PHP lib or on the server? Maybe I can track it down if it's in the PHP lib. I'm using `captureException` instead of the registering the error handler.

**Comment by mnpenner** [comment]
I traced it down to `sentry/sentry/lib/Raven/Stacktrace.php` line 97:

```php
            // detect in_app based on app path
            if ($app_path) {
                $data['in_app'] = (bool)(substr($abs_path, 0, strlen($app_path)) === $app_path);
            }
```

That's coming back as `true` so it *is* flagging my internal stuff as "in_app", but it's not showing that way in the web app. Is that a new feature? Is it available in the self-hosted version?

**Comment by dcramer** [comment]
@mnpenner good catch on trailing slash -- we def need to fix that. 

If all frames are in_app then we actually disable the split frames feature. If that sounds like the case here, then we'll chock this one up to needing to improve that detection.

**Comment by torotil** [comment]
Could you make the connection between `in_app` and the original issue more explicit?

I have the same issue with a Drupal 7 site. In this case all the code (including the sentry-php library) resides below Drupal's htdocs folder. How would I get clean stack-traces in this case? Is there a way to include/exclude paths? In this case everything in `/var/www/drupal7/htdocs/` would be in_app but everything in `/var/www/drupal7/htdocs/sites/all/modules/raven/` or `/var/www/drupal7/htdocs/sites/all/libraries/sentry-php` wouldn't be.

Drupal's own error reporting has a blacklist of functions that are removed from the stack in any case. This could be a way to make that work in sentry too. Although dividing the stack as mentioned above sounds more superior since it doesn't lose information.

**Comment by dcramer** [comment]
@torotil in the Python SDK we have exclude_paths -- they're used to ensure files aren't marked as in_app. We could potentially do something like that here. One complexity in the PHP SDK is we decided it would be simpler to have a single app path (going back to Python, it has a list of include_paths). That seems fine in the Drupal example as long as there were a way to exclude things.

**Comment by torotil** [comment]
Thanks for explaining! If we had excludes the PHP SDK could also exclude itself by default.

**Comment by dcramer** [comment]
The question comes down to what a good descriptive name is for the setting:

- app_path_exclude?
- exclude_app_paths?
- ???

**Comment by torotil** [comment]
I'd vote for `app_excluded_paths` ;)

**Comment by stayallive** [comment]
Looking at the current options I would vote for `excluded_app_paths`.

**Comment by ste93cry** [comment]
The issue is quite old and should be hopefully fixed in `2.0`, so I'm going to close this. Feel free to open a new issue if it's still the case
